### PR TITLE
feat(STONEINTG-459): remove cache from controller unittest

### PR DESCRIPTION
**Pull request by hongweiliu17**
Signed-off-by: Hongwei Liu <hongliu@redhat.com>

### CODE REVIEW

File: controllers/snapshot/snapshot_controller.go [hunk 1]
```diff
@@ -133,6 +133,10 @@ func setupCache(mgr ctrl.Manager) error {
 		return err
 	}
 
+	if err := cache.SetupIntegrationTestScenarioCache(mgr); err != nil {
```
[the remark this answers is not in the document]
Comment (dirgim) [reply controllers/snapshot/snapshot_controller.go:136]: My mistake, my above comment is not relevant, that was already handled correctly in the previous PR.

File: controllers/snapshot/snapshot_controller_test.go [hunk 2]
```diff
@@ -140,6 +140,12 @@ var _ = Describe("SnapshotController", func() {
 	})
 
 	It("can Reconcile function prepare the adapter and return the result of the reconcile handling operation", func() {
+		req := ctrl.Request{
```
Comment (hongweiliu17) [inline controllers/snapshot/snapshot_controller_test.go:143]: If we set the `req` to a non-existing CR and remove `Expect(setupCache(k8sManager)).To(Succeed())` and some cache functions, the case can pass. But some code will not be covered ([coverage report](https://app.codecov.io/gh/redhat-appstudio/integration-service/pull/225/blob/controllers/snapshot/snapshot_controller.go)), release-service coverage [here](https://app.codecov.io/gh/redhat-appstudio/release-service/blob/main/controllers%2Frelease%2Fcontroller.go#L79) is similar.

If we use a non-existing CR and remove `Expect(setupCache(k8sManager)).To(Succeed())` from other _suite_test.go, they are similar.

Can we do like this? Thanks!

Comment (dirgim) [reply controllers/snapshot/snapshot_controller_test.go:143]: Yes, I think that your proposed change is reasonable.

**Review by dirgim — APPROVED** [review]
LGTM

File: controllers/snapshot/snapshot_controller.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (dirgim) [reply controllers/snapshot/snapshot_controller.go]: Good point, that seems to have resolved it.

**Comment by hongweiliu17** [comment]
Rebase code and remove `Expect(setupCache(k8sManager)).To(Succeed())` from *_suite_test.go and add mocked data caused by remove `setupCache`.

**Review by dheerajodha — APPROVED** [review]
LGTM

**Review by jinqi7 — APPROVED** [review]
LGTM

**Pull request merged**